mkosi/__init__.py

```diff
@@ -2493,8 +2494,35 @@ def invoke_apt(
         INITRD="No",
     )
 
-    run_workspace_command(args, root, cmdline, network=True, env=env)
+    # Ensure apt inside the image can see the local mirror outside of it
+    if args.local_mirror:
+        mirror : Optional[str] = args.local_mirror
```

Comment:
This cannot be `Optional[str]`, because the check in the line before will ensure `args.local_mirror` is not `None` and not the empty string, so the line conditional can be condensed to
```python
mirror = args.local_mirror or args.mirror
```

Reply:
I've done this, but now mypy fails with:

```
mkosi/__init__.py:2498:8: error: Item "None" of "Optional[str]" has no
attribute "startswith"  [union-attr]
        if mirror.startswith("file:"):
           ^
mkosi/__init__.py:2499:26: error: Value of type "Optional[str]" is not
indexable  [index]
            nspawn_params = [f"--bind-ro={mirror[[5](https://github.com/systemd/mkosi/runs/7970986764?check_suite_focus=true#step:7:6):]}"]
```

Reply:
Ah, my bad I misremembered `args.mirror` as `str`, but it is also `Optional[Str]` so the check needs to stay
```python
if mirror and mirror.startswith("file:")
```